### PR TITLE
chore: standardize npm script names

### PR DESCRIPTION
**Pull request by snitin315**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request?

Refers https://github.com/eslint/eslint/issues/14827

#### What changes did you make? (Give an overview)

This updates the names of the scripts in package.json to be consistent with the [new standard](https://eslint.org/docs/latest/contribute/package-json-conventions).

#### Is there anything you'd like reviewers to focus on?
We may need to update the `release` related scripts on Jenkins before merging this PR.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,13 +17,13 @@
   ],
   "scripts": {
     "lint": "eslint .",
-    "test": "mocha ./tests/lib/**/*.js",
     "new-rule-format": "node ./bin/eslint-transforms.js -t ./lib/new-rule-format/new-rule-format.js",
```
Comment (mdjermanovic) [inline package.json:20]: Any ideas on what this script is doing? Per the [new standard](https://eslint.org/docs/latest/contribute/package-json-conventions#main-script-names), with the exception of npm life cycle scripts all scripts should begin with `build:`, `release:`, `lint:`, `start:`, or `test:`.

Comment (snitin315) [reply package.json:20]: I think we can prefix it with `build:`, it seems to be [updating rule from the old format to new format](https://github.com/eslint/eslint-transforms/blob/b044f93a74d7f1193878f2fadbaf57a3509761d6/lib/new-rule-format/new-rule-format.js#L2)

Comment (snitin315) [reply package.json]: Hmm, it doesn't seem to fit any of the `build:`, `release:`, `lint:`, `start:`, or `test:` categories. Should we keep it as-is?

Comment (nzakas) [reply package.json]: I think we can just remove it.

Comment (mdjermanovic) [reply package.json]: > I think we can just remove it.

Agreed.

Comment (snitin315) [reply package.json]: removed.

**Review by nzakas — APPROVED** [review]
LGTM. Would like @mdjermanovic to verify before merging.

**Comment by mdjermanovic** [comment]
> We may need to update the `release` related scripts on Jenkins before merging this PR.

Updated.

**Pull request merged**